overlord/snapstate/snapmgr.go

```diff
@@ -183,6 +184,30 @@ func (snapsup *SnapSetup) MountFile() string {
 	return snap.MountFileInDir(blobDir, snapsup.InstanceName(), snapsup.Revision())
 }
 
+// ComponentSetup holds the necessary component details to perform
+// most component related tasks.
+type ComponentSetup struct {
+	// CompSideInfo for metadata not coming from the component
+	CompSideInfo *snap.ComponentSideInfo `json:"comp-side-info,omitempty"`
+	// CompPath is the path to the file
+	CompPath string `json:"comp-path,omitempty"`
+
+	// SnapSup contains information about the owner snap
+	SnapSup *SnapSetup `json:"snap-setup,omitempty"`
```

Comment:
I meant as a separate data field on the tasks, not inside ComponentSetup, so we get the logic from affectedSnaps in conflict.go for example

Reply:
Changed now